sidechain/block-composer/src/block_composer.rs

```diff
@@ -148,11 +154,19 @@ where
 			now_as_u64(),
 		);
 
+		let mut finalization_candidate = next_finalization_block_number;
+		if block_number == 1 {
+			finalization_candidate = 1;
```

Comment:
This is wrong, and shouldn't this fail after your fixes in the pallet yesterday?

Reply:
this is necessary becasue we want to check block_number == next_finalization_block_number (PREVIOUS next_finalization_block_number)
Then I have to add BLOCK_NUMBER_FINALIZATION_DIFF later, before we send it. But this is for the NEXT next_finalization_block_number. 

The alternative would be to to add it before, but then the check would block_
`number == next_finalization_block_number  - BLOCK_NUMBER_FINALIZATION_DIFF` but this is incredibly more ugly.

Reply:
Ah, I understand now. Hmm, I can see that this is necessary with the current flow, although I am not 100% happy with it. We might come up with a more elegant/intuitive design in the future - but not now. :)